ext/include/opentelemetry/ext/zpages/tracez_data_aggregator.h

```diff
@@ -0,0 +1,127 @@
+#pragma once
+
+// include libraries
+#include <string>
+#include <map>
+#include <unordered_set>
+#include <vector>
+#include <list>
+
+// include files
+#include "opentelemetry/ext/zpages/tracez_processor.h"
+#include "opentelemetry/sdk/trace/span_data.h"
+#include "opentelemetry/ext/zpages/latency_boundaries.h"
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/trace/canonical_code.h"
+
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace ext
+{
+namespace zpages
+{
+
+/**
+ * kMaxNumberOfSampleSpans is the maximum number of error and latency sample spans stored at any given time.
+ */
+const int kMaxNumberOfSampleSpans = 5;
+
+/**
+ * AggregatedInformation is the aggregated span information that is stored for each span name. 
+ */
+struct AggregatedInformation{
+  
+  int running_spans_; 
+  int error_spans_;
+  
+  /** 
+   * latency_sample_spans_ is a vector of lists, each index of the vector corresponds to a latency boundary(of which there are 9).
+   * The list in each index stores the sample spans for that latency boundary.
+   */
+  std::vector<std::list<std::unique_ptr<opentelemetry::sdk::trace::Recordable>>> latency_sample_spans_;
```

Comment:
Is there a specific reason why a vector is needed (over an array), given that the size is fixed?

Reply:
No particular reason, I wasn't sure if it was going to be fixed initially. I can go in and change this but since the size is 9, I would not be too worried about the memory efficiency that an array gives me

Reply:
On second thought, I think I am going to go in and change it